### PR TITLE
Run tests with ext-pgsql

### PR DESCRIPTION
**Pull request by derrabus**
A new CI job that runs our test suite with the new pgsql driver introduced in doctrine/dbal#5880.

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -116,13 +116,21 @@ jobs:
...
           - php-version: "8.0"
             dbal-version: "2.13"
             postgres-version: "14"
+            extension: pdo_pgsql
           - php-version: "8.2"
             dbal-version: "default"
             postgres-version: "9.6"
+            extension: pdo_pgsql
+          - php-version: "8.2"
+            dbal-version: "3@dev"
```
Comment (greg0ire) [inline .github/workflows/continuous-integration.yml]: Let's wait for a release of DBAL and use a stable version? That way you're unlikely to break the ORM build for nothing, and there is no need to make another PR to alter this.

Comment (derrabus) [reply .github/workflows/continuous-integration.yml]: All right.

**Review by greg0ire — APPROVED** [review]
